src/backend/distributed/commands/utility_hook.c

```diff
@@ -656,6 +657,30 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 	 */
 	if (EnableDDLPropagation)
 	{
+		/*
```

Comment:
can we move this code block to a function like: `ErrorIfCircularDependencyExistsAtPostpress(ops, parsetree)`?